apps/admin_api/lib/admin_api/v1/plugs/user_auth_plug.ex

```diff
@@ -18,29 +18,48 @@ defmodule AdminAPI.V1.UserAuthPlug do
 
   def init(opts), do: opts
 
-  def call(conn, _opts) do
+  def call(conn, opts) do
+    auth = Keyword.get(opts, :enable_client_auth,
```

Comment:
Can this go into `init/1` instead? Because it looks like option initialization to me. E.g.

```ex
  def init(opts) do
    Keyword.put_new(opts, :enable_client_auth, Application.get_env(:admin_api, :enable_client_auth))
  end

  def call(conn, opts) do
    if opts[:enable_client_auth] do
      conn
      |> parse_header()
      |> authenticate()
    else
      assign(conn, :authenticated, :client)
    end
  end
```